Yosemite/Yosemite/Base/Dispatcher.swift

```diff
@@ -69,7 +69,12 @@ public class Dispatcher {
     public func dispatch(_ action: Action) {
         assertMainThread()
 
-        processors[action.identifier]?.onAction(action)
+        // Avoid silent failure when a store is not retained
+        guard let processor = processors[action.identifier] else {
+            assertionFailure("⛔️ No processor found for \(action.identifier)!")
+            return
```

Comment:
Pro tip: Since `assertion failure` returns `Void` under the hood, you can do
```suggestion
            return assertionFailure("⛔️ No processor found for \(action.identifier)!")

```

And keep your guard statements more concise!